### DISCUSSION TITLE
Can't interact by network with bitbucket, is this intentional?

### DISCUSSION THREAD

**Discussion by bigcat88** [cirruslabs/cirrus-ci-docs#1130] [Q&A] [answered]
Trying to move build process of a python library here, and in the repository I have code that can be described as:
```
curl -L https://bitbucket.org/multicoreware/x265_git/get/0b75c44c10e605fe9e9ebed58f04a46271131827.tar.gz -o download.tar.gz
```
to download `x265` encoder for build...

But I can't ping `bitbucket.org`, `git clone` fails, so the question is, what should I do to make it work?
For me it looks like a firewall issue.

**Comment by fkorotkov** [comment, marked as answer]
Hey @bigcat88, yes, this is intentional. We've seen a lot of Crypto miners using BitBucket, Gitea, GitLab and others for hosting the scripts and mining software. Kind of using them as "control plane" and than abusing Cirrus CI and other CIs.

In your case you can try to download sources from http://ftp.videolan.org/pub/videolan/x265/ to workaround the issue. Sorry for the inconveniences!

**Reply by bigcat88** [reply]
I can't do that, I need latest version(sometimes not last, because they often break build, currently I have pinned commit with successful build on all OS & CPU), not a two year old that has bug on arm64.

**Reply by fkorotkov** [reply]
I checked Firewall rules and logs. Seems we haven't seen crypto mining attempts using BitBucket in a long time so I've disable the Firewall rule for now. We might not need it at all since we've implemented couple more mechanism for preventing crypto mining since then.
